src/internal.c

```diff
@@ -21543,9 +21579,18 @@ static int GetInputData(WOLFSSL *ssl, word32 size)
             return RECV_OVERFLOW_E;
         }
 
+        if ((word32)in < size) {
+            if (!DtlsRecordsCanSpanDatagrams(ssl)) {
+                WOLFSSL_MSG("DTLS: Received partial record, ignoring");
+#ifdef WOLFSSL_DTLS_DROP_STATS
+                ssl->replayDropCount++;
+#endif /* WOLFSSL_DTLS_DROP_STATS */
+                continue;
+            }
+        }
+
         ssl->buffers.inputBuffer.length += (word32)in;
```

Comment:
I don't know if dropping and continuing to read is the best option. I think we should check if we are non-blocking. If we are non-blocking then return `WANT_READ`. Also this section needs to be protected with `IsDtlsNotSctpMode`.

Reply:
I think the end result will be the same as the next invocation of `wolfSSLReceive` will return `WANT_READ` in that case. I would like to avoid returning `WANT_READ` outside of the real networking function.  
`DtlsRecordsCanSpanDatagrams` returns always 1 for the `IsDtlsNotSctpMode`

Reply:
`DtlsRecordsCanSpanDatagrams` implies that this is implemented for DTLS only. If you don't want to protect the call with dtls checks I would suggest changing the name to just `RecordsCanSpanMultipleReads` or something simpler.

Reply:
renamed and simplified nested ifdef